### PR TITLE
Show flag name instead of True/False with Boolean flags

### LINKED ISSUE #1538: Improvement for show_default showing in case of a flag the name instead of True/False
First to say: It's not a bug ... it's (in my opinion) a reasonable improvement.

### Expected Behavior

That's how the boolean flag is configured:

```python
@click.option('--cache/--no-cache', default=False, show_default=True,
              help="Enable/Diable the cache.")
```
Using the --help you see following (just the line of interest):

```
--cache / --no-cache        Enable/Diable the cache.  [default: False]
```

but preferable woud be:

```
--cache / --no-cache        Enable/Diable the cache.  [default: no-cache]
```

It's clear to me that this works for --cache/--no-cache like flags only.

### Actual Behavior

The output is as documented by the API the default **value** (*not the name*) which is for a boolean True or False.

### Environment

* Python version: 3.8.1
* Click version: 7.0

### PR DESCRIPTION
**Pull request by ovezovs**
Fixes #1538

### CODE REVIEW

**Comment by ovezovs** [comment]
@davidism, I have one major question regarding this fix: Should we show the flag names only for boolean flags or any? For example, [test_global_show_default](https://github.com/pallets/click/blob/fb1ca91ab00a2ce12da84addc459e1983aac2fcf/tests/test_formatting.py#L303) expects default on the help flag line to be false. For now, I did a workaround to not display the default flag name when it is help. I am not sure if that's a bad thing.

File: src/click/core.py [hunk 1]
```diff
@@ -1939,6 +1939,10 @@ def _write_opts(opts):
                 default_string = "(dynamic)"
             else:
                 default_string = self.default
+                if self.is_bool_flag and self.name != "help":
+                    default_string = (
+                        split_opt(rv[0])[1] if self.default else split_opt(rv[1])[1]
```
Comment (juped) [inline src/click/core.py:1944]: Maybe this line could use a short comment

**Comment by davidism** [comment]
Unless the test is particularly complex, I usually just roll it into the code commit. Same for multiple code and cleanup commits into a single commit.

**Comment by davidism** [comment]
Changed the implementation so that it checks that the boolean flag actually has separate flags for `True` and `False`. It selects them from `self.opts` and `self.secondary_opts` instead of `rv`, since `rv` will have `,` joined alternative names together. That would cause the following to output `[default: no-cache, --nc]`.

```python
@click.option("--cache/--no-cache", "--c/--nc", show_default=True)
```

Updated the test to check when `default` is `True` and `False`, and added a second test for when there's only one name.

**Comment by ovezovs** [comment]
> Updated the implementation so that it checks that the boolean flag actually has separate flags for `True` and `False`. It selects them from `self.opts` and `self.secondary_opts` instead of `rv`, since `rv` will have `,` joined alternative names together. That would cause the following:
> 
> ```python
> @click.option("--cache/--no-cache", "--c/--nc", show_default=True)
> ```
> 
> ```
> [default: no-cache, --nc]
> ```
> 
> Now it outputs only `no-cache`.
> 
> Updated the test to check when `default` is `True` and `False`, and added a second test for when there's only one name.

Thank you so much for the feedback!

**Pull request merged**